libs/SmartSync/SmartSyncTests/SyncManagerTests.m

```diff
@@ -1183,6 +1183,30 @@ - (void) testReSyncRunningSync
     while ([queue getNextSyncUpdate].status != SFSyncStateStatusDone);
 }
 
+- (void) testSyncForSelectedPath {
+    [self createAccountsSoup:ACCOUNTS_SOUP];
+    
+    NSString* soql = @"SELECT Id, Name, BillingCity, LastModifiedDate FROM Account limit 10";
+    SFSoqlSyncDownTarget* target = [SFSoqlSyncDownTarget newSyncTarget:soql];
+    
+    SFSyncOptions* options = [SFSyncOptions newSyncOptionsForSyncDown:SFSyncStateMergeModeOverwrite];
+    SFSyncState* sync = [SFSyncState newSyncDownWithOptions:options target:target soupName:ACCOUNTS_SOUP store:store];
+    // Runs sync.
+    SFSyncUpdateCallbackQueue* queue = [[SFSyncUpdateCallbackQueue alloc] init];
+    [queue runSync:sync syncManager:syncManager];
+    while (true) {
+        SFSyncState *state = [queue getNextSyncUpdate];
+        if (SFSyncStateStatusDone == state.status || SFSyncStateStatusFailed == state.status) {
+            XCTAssertEqual(SFSyncStateStatusDone, state.status, @"Expect Sync Successfully");            
+            break;
+        }
+    }
+    
+    SFQuerySpec *querySpec = [SFQuerySpec newMatchQuerySpec:ACCOUNTS_SOUP withSelectPaths:@[ACCOUNT_NAME,ACCOUNT_ID] withPath:ACCOUNT_NAME withMatchKey:@"Acme" withOrderPath:ACCOUNT_NAME withOrder:kSFSoupQuerySortOrderDescending withPageSize:10];
+    
+    NSArray *result =[store queryWithQuerySpec:querySpec pageIndex:0 error:nil];
+    XCTAssertNotNil(result);
```

Comment:
@wmathurin  I am trying to add this test for basic select path E2E test, however, keep hitting an error of as below.. will you please help advice what I missed? Really appreciated!  @helenren @kchitalia
2016/06/06 20:37:41.640 SmartSyncTestApp[89292:5c07] ERROR com.salesforce <SF_FMDatabase>: DB Error: 1 "no such table: TABLE_2_fts"
2016/06/06 20:37:41.640